crates/sound/src/audio_backends/alsa.rs

```diff
@@ -0,0 +1,550 @@
+                    virtio_sound::VIRTIO_SND_PCM_RATE_32000 => 32000,
+                    virtio_sound::VIRTIO_SND_PCM_RATE_44100 => 44100,
+                    virtio_sound::VIRTIO_SND_PCM_RATE_48000 => 48000,
+                    virtio_sound::VIRTIO_SND_PCM_RATE_64000 => 64000,
+                    virtio_sound::VIRTIO_SND_PCM_RATE_88200 => 88200,
+                    virtio_sound::VIRTIO_SND_PCM_RATE_96000 => 96000,
+                    virtio_sound::VIRTIO_SND_PCM_RATE_176400 => 176400,
+                    virtio_sound::VIRTIO_SND_PCM_RATE_192000 => 192000,
+                    virtio_sound::VIRTIO_SND_PCM_RATE_384000 => 384000,
+                    _ => 44100,
+                },
+                ValueOr::Nearest,
+            )?;
+            hwp.set_format(match s.params.format {
+                virtio_sound::VIRTIO_SND_PCM_FMT_IMA_ADPCM => Format::ImaAdPCM,
+                virtio_sound::VIRTIO_SND_PCM_FMT_MU_LAW => Format::MuLaw,
+                virtio_sound::VIRTIO_SND_PCM_FMT_A_LAW => Format::ALaw,
+                virtio_sound::VIRTIO_SND_PCM_FMT_S8 => Format::S8,
+                virtio_sound::VIRTIO_SND_PCM_FMT_U8 => Format::U8,
+                virtio_sound::VIRTIO_SND_PCM_FMT_S16 => Format::s16(),
+                virtio_sound::VIRTIO_SND_PCM_FMT_U16 => Format::r#u16(),
+                virtio_sound::VIRTIO_SND_PCM_FMT_S18_3 => Format::S183LE,
+                virtio_sound::VIRTIO_SND_PCM_FMT_U18_3 => Format::U183LE,
+                virtio_sound::VIRTIO_SND_PCM_FMT_S20_3 => Format::S203LE,
+                virtio_sound::VIRTIO_SND_PCM_FMT_U20_3 => Format::U203LE,
+                virtio_sound::VIRTIO_SND_PCM_FMT_S24_3 => Format::S24LE,
+                virtio_sound::VIRTIO_SND_PCM_FMT_U24_3 => Format::U24LE,
+                virtio_sound::VIRTIO_SND_PCM_FMT_S20 => Format::s20_3(),
+                virtio_sound::VIRTIO_SND_PCM_FMT_U20 => Format::u20_3(),
+                virtio_sound::VIRTIO_SND_PCM_FMT_S24 => Format::s24(),
+                virtio_sound::VIRTIO_SND_PCM_FMT_U24 => Format::u24(),
+                virtio_sound::VIRTIO_SND_PCM_FMT_S32 => Format::s32(),
+                virtio_sound::VIRTIO_SND_PCM_FMT_U32 => Format::r#u32(),
+                virtio_sound::VIRTIO_SND_PCM_FMT_FLOAT => Format::float(),
+                virtio_sound::VIRTIO_SND_PCM_FMT_FLOAT64 => Format::float64(),
+                virtio_sound::VIRTIO_SND_PCM_FMT_DSD_U8 => Format::DSDU8,
+                virtio_sound::VIRTIO_SND_PCM_FMT_DSD_U16 => Format::DSDU16LE,
+                virtio_sound::VIRTIO_SND_PCM_FMT_DSD_U32 => Format::DSDU32LE,
+                virtio_sound::VIRTIO_SND_PCM_FMT_IEC958_SUBFRAME => Format::iec958_subframe(),
+                _ => Format::Unknown,
+            })?;
+
+            hwp.set_access(Access::RWInterleaved)?;
+
+            // > A period is the number of frames in between each hardware interrupt.
+            // - https://www.alsa-project.org/wiki/FramesPeriods
+
+            // FIXME
```

Comment:
What are those FIXMEs are about? Should drop the parts that are no longer relevant or properly document what is needed to be fixed.

Reply:
Wrote down what the FIXME was about and removed the commented code.

Reply:
(Feel free to resolve the conversation if the added comment is adequate)

Reply:
Apparently I can't resolve the conversation myself. But lgtm